src/main/java/com/google/sps/data/Section.java

```diff
@@ -14,57 +14,94 @@
-        return false;
+    for (int i = 0; i < meetingTimes.size() - 1; i++) {
+      if (meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
+        throw new IllegalArgumentException("A section's lecture times can't overlap!");
       }
+    }
+  }
 
-      Section otherSection = (Section) other;
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
+    }
 
-      return otherSection.professor.equals(this.professor)
-              && otherSection.meetingTimes.equals(this.meetingTimes);
+    if (!(other instanceof Section)) {
+      return false;
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(professor, meetingTimes);
+    Section otherSection = (Section) other;
+
+    return otherSection.professor.equals(this.professor) 
+      && otherSection.meetingTimes.equals(this.meetingTimes);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(professor, meetingTimes);
+  }
+
+  public List<TimeRange> getMeetingTimes() {
+    return this.meetingTimes;
+  }
+
+  /**
+   * Function to return whether or not a section overlaps with another.
+   * 
+   * @param other the section to check for overlap with.
+   * @return If there's overlap between the current section and the other section
+   */
+  public boolean overlaps(Section other) {
+    // A section with no meeting times can't overlap with another section.
+    if (this.meetingTimes.isEmpty() || other.meetingTimes.isEmpty()) {
+      return false;
     }
 
+    ArrayList<TimeRange> combinedTimes = new ArrayList<>(this.meetingTimes);
+    combinedTimes.addAll(other.meetingTimes);
 
-    /**
-     * Function to return whether or not a section overlaps with another
-     * @param other
-     * @return If there's overlap between the current section and the other section
-     */
-    public boolean overlaps(Section other){
-        //TODO: This
-        return false;
+    Collections.sort(combinedTimes, TimeRange.ORDER_BY_START);
+    int timeCount = combinedTimes.size();
+
+    for (int i = 0; i < timeCount - 1; i++) {
+      if (combinedTimes.get(i).overlaps(combinedTimes.get(i + 1))) {
+        return true;
+      }
     }
-}
\ No newline at end of file
+
+    TimeRange lastEvent = combinedTimes.get(timeCount - 1);
+    return (lastEvent.end() > TimeRange.END_OF_WEEK) 
+      && saturdayOverlapsSunday(lastEvent, combinedTimes.get(0));
+  }
+
+  /**
+   * A helper function that checks if a TimeRange that starts on Saturday but ends on Sunday 
+   * overlaps with a TimeRange earlier in the week.
+   * 
+   * @param satEvent the event that starts on Sat but ends on Sunday
+   * @param earliestEvent the event earlier in the week.
+   * 
+   * @return a boolean representing if the two events overlap.
+   */
+  private boolean saturdayOverlapsSunday(TimeRange satEvent, TimeRange earliestEvent) {
+    return earliestEvent.contains(satEvent.end() - TimeRange.END_OF_WEEK);
+  }
```

Comment:
Re-iterating this here so that it doesn't get lost in the other comment chain; this behavior belongs in `TimeRange`, not in `Section`.

Reply:
I didn't see those other comments! Thanks.

Reply:
Fixed!